marlowe-dashboard-client/src/MainFrame/View.purs

```diff
@@ -1,165 +1,240 @@
+            ]
+            [ Icon.contacts
+            , span
+                [ classNames [ "ml-0.5" ] ]
+                [ text "Contacts" ]
+            ]
+        , a
+            [ classNames itemClasses
+            , onClick $ const $ Just $ ToggleCard PutdownWalletCard
+            ]
+            [ Icon.wallet
+            , span
+                [ classNames [ "ml-0.5", "hidden", "md:inline" ] ]
+                [ text "Wallet" ]
+            ]
+        , a
+            [ classNames $ itemClasses <> [ "md:hidden" ]
+            , onClick $ const $ Just ToggleMenu
+            ]
+            [ if menuOpen then Icon.close else Icon.menu ]
+        , a
+            [ classNames $ itemClasses <> [ "px-1", "bg-green", "text-white" ]
+            , onClick $ const $ Just $ ToggleCard TemplateLibraryCard
             ]
             [ span
-                [ classNames [ "px-0.5" ] ]
+                [ classNames [ "mr-0.5" ] ]
                 [ text "New" ]
             , Icon.libraryAdd
             ]
         ]
     ]
   where
-  menuIcon = if currentOverlay == Just Menu then Icon.close else Icon.menu
+  itemClasses = [ "flex", "items-center", "p-0.5" ]
 
-  navButton icon action =
-    button
-      [ classNames $ buttonClasses <> [ "text-green" ]
-      , onClick $ const $ action
-      ]
-      [ icon ]
+renderMain :: forall p. WalletNicknameKey -> PubKeyHash -> WalletLibrary -> Boolean -> Screen -> Maybe Card -> HTML p Action
+renderMain newWalletNicknameKey wallet wallets menuOpen screen card =
+  main
+    [ classNames [ "relative", "bg-lightblue", "text-blue" ] ]
+    [ renderMobileMenu menuOpen
+    , renderCards newWalletNicknameKey wallets wallet card
+    , renderScreen wallets screen
+    ]
 
-menu :: forall m. MonadAff m => Maybe Overlay -> Screen -> ComponentHTML Action ChildSlots m
-menu currentOverlay currentScreen =
+renderMobileMenu :: forall p. Boolean -> HTML p Action
+renderMobileMenu menuOpen =
   div
-    [ wrapperClassNames ]
+    [ classNames $ [ "md:hidden", "absolute", "top-0", "bottom-0", "left-0", "right-0", "z-20", "bg-transgray" ] <> hideWhen (not menuOpen) ]
     [ nav
-        [ classNames [ "bg-gray", "mx-0.5", "mb-0.5", "flex", "flex-col", "justify-between" ] ]
-        [ nav
+        [ classNames $ [ "absolute", "top-0", "bottom-0.5", "left-0.5", "right-0.5", "bg-gray", "overflow-auto", "flex", "flex-col", "justify-between" ] ]
+        [ div
             [ classNames [ "flex", "flex-col" ] ]
-            [ menuItem "Dashboard home" $ Just $ SetScreen Home
-            , menuItem "Contacts" $ Just $ SetScreen Contacts
-            , menuItem "marlowe.io" Nothing
-            , menuItem "Library" Nothing
-            , menuItem "Docs" Nothing
-            , menuItem "Support" Nothing
+            [ link "Dashboard home" $ Right $ SetScreen $ ContractsScreen Running
+            , link "Contacts" $ Right $ SetScreen WalletLibraryScreen
+            , link "Library" $ Left ""
```

Comment:
The empty "" means that we don't have an URL for those yet? If so we can put a [Warn](https://pursuit.purescript.org/builtins/docs/Prim.TypeError#t:Warn) in the function so we don't forget

Reply:
👍